lib/types.nix

```diff
@@ -326,110 +326,15 @@ rec {
-                fallback = { name = "name"; };
-              in findFirst ({ path, ... }: all (v: v == true) (zipListsWith equals loc path)) fallback nameAttrs;
-              nameAttr = matched.name;
-              nameValueOld = value:
-                if isList value then
-                  if length value > 0 then
-                    "[ " + concatMapStringsSep " " escapeNixString value + " ]"
-                  else
-                    "[ ]"
-                else
-                  escapeNixString value;
-              nameValueNew = value: unnamed:
-                if isList value then
-                  if length value > 0 then
-                    head value
-                  else
-                    unnamed
-                else
-                  value;
-              res =
-                { inherit (def) file;
-                  value = listToAttrs (
-                    imap1 (elemIdx: elem:
-                      { name  = nameValueNew (elem.${nameAttr} or (unnamed elemIdx)) (unnamed elemIdx);
-                        value = elem;
-                      }) def.value);
-                };
-              option = concatStringsSep "." loc;
-              sample = take 3 def.value;
-              more = lib.optionalString (length def.value > 3) "... ";
-              list = concatMapStrings (x: ''{ ${nameAttr} = ${nameValueOld (x.${nameAttr} or "unnamed")}; ...} '') sample;
-              set = concatMapStrings (x: ''${nameValueNew (x.${nameAttr} or "unnamed") "unnamed"} = {...}; '') sample;
-              msg = ''
-                In file ${def.file}
-                a list is being assigned to the option config.${option}.
-                This will soon be an error as type loaOf is deprecated.
-                See https://github.com/NixOS/nixpkgs/pull/63103 for more information.
-                Do
-                  ${option} =
-                    { ${set}${more}}
-                instead of
-                  ${option} =
-                    [ ${list}${more}]
-              '';
-            in
-              lib.warn msg res
-          else
-            def;
-        attrOnly = attrsOf elemType;
-      in mkOptionType rec {
-        name = "loaOf";
-        description = "list or attribute set of ${elemType.description}s";
-        check = x: isList x || isAttrs x;
-        merge = loc: defs: attrOnly.merge loc (convertAllLists loc defs);
-        emptyValue = { value = {}; };
-        getSubOptions = prefix: elemType.getSubOptions (prefix ++ ["<name?>"]);
-        getSubModules = elemType.getSubModules;
-        substSubModules = m: loaOf (elemType.substSubModules m);
-        functor = (defaultFunctor name) // { wrapped = elemType; };
-      };
+    # drop this in the future:
+    loaOf =
+      let msg =
+        ''
+          `types.loaOf` is deprecated and mixing list with attribute values
+          is no longer possible; use only `types.attrsOf` instead.
+          See https://git.io/JUvd1 for the motivation.
```

Comment:
```suggestion
          `types.loaOf` has been removed and mixing lists with attribute values
          is no longer possible; please use `types.attrsOf` instead.
          See https://git.io/JUvd1 for the motivation.
```

Even though `types.loaOf` is aliased to `types.attrsOf` now, I still think this counts as being removed. However, if this wording isn't desired, I think at least the following minor changes should be included:

```suggestion
          `types.loaOf` is deprecated and mixing lists with attribute values
          is no longer possible; please use `types.attrsOf` instead.
          See https://git.io/JUvd1 for the motivation.
```

Reply:
Thanks for the suggestions. The rewording looks good to me: I've implemented it.

Reply:
Could we not use URL shorteners? They obscure the target location and in this case the URL is not even that long.

Reply:
Switched to original link.